subsys/bluetooth/host/keys.c

```diff
@@ -371,3 +412,26 @@ SETTINGS_STATIC_HANDLER_DEFINE(bt_keys, "bt/keys", NULL, keys_set, keys_commit,
 			       NULL);
 
 #endif /* CONFIG_BT_SETTINGS */
+
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+void bt_keys_notify_usage(u8_t id, const bt_addr_le_t *addr)
+{
+	if (!addr) {
+		return;
+	}
+	struct bt_keys *keys = bt_keys_find_addr(id, addr);
```

Comment:
Put the variable declaration in the beginning of the function, and then here you'd have:
```
keys = bt_keys_find_addr(id, addr);
if (!keys) {
        return;
}

...
```

Reply:
We can not call `bt_keys_find_addr` with addr=NULL. We do not expect NULL here, maybe remove the addr verification?